woocommerce/woocommerce-ios
# Fix Product images order

"fix the order of Product images (requires schema migration)" step in #1713 

## Summary

The order of the Product images matters because it is the order they appear on the production Shop > Product page. Before this PR, `Product`'s `images` is an unordered set when we import the images from the API response to Storage and thus the order is lost. This PR preserves the order of Product images to be the same as the order in the API response.

## Changes

Note: the actual changes are after the first commit that created the model version, I'd recommend reviewing by each commit

- Created a new model version 26 based on 25
- Updated the `Product`'s `images` relationship to be ordered, and now the `images` property is an `NSOrderedSet`
- Created an extension for `NSOrderedSet` to return a generic array as a helper
- In `ProductStore`, imported the images in order
- In `extension Storage.Product: ReadOnlyConvertible`, removed the default sorting since the order is already defined in `Product`'s `images`

## Testing

Prerequisite: have a Product with multiple images

- Build and launch the app from the latest `develop`, and just tap each tab to make sure the data are showing correctly. Notice the Product images might not be the same order as in `wp-admin`
- Build and launch the app from the PR branch --> make sure the app doesn't crash, and the Product images are in the same order as in `wp-admin`

Update release notes:
